datastore/datastore.go

```diff
@@ -58,7 +64,10 @@ func New() EntityStore {
 }
 
 type VersionedEntity struct {
-	DatabaseVersion int `json:",omitempty"`
+	DatabaseVersion int    `json:"_version,omitempty"`
+	IfSeqNo         int    `json:"_if_seq_no,omitempty"`
+	IfPrimaryTerm   int    `json:"_if_primary_term,omitempty"`
+	Type            string `json:"type,omitempty"`
```

Comment:
The _if_seq_no and _if_primary_term fields are the new mechanisms of versioning in new ES. The changes in _version field are illegal now. 
The type was included in the entity, because of depreciation of the mapping types mechanism